### DISCUSSION TITLE
H100 vs A100 GPU

### DISCUSSION THREAD

**Discussion by TomPaolo** [BLAST-WarpX/warpx#4415] [Installation] [answered]
Dear developers, 
we have WarpX running on our small 4xA100 80Gb node, but we are about to invest on a decently big GPU cluster (32-40 GPU). 
We were oriented towards the H100 80Gb cards. You tested the performances on V100 and A100 systems (Parallel Computing 108), we wonder if you tested on the H100 cards, too, as we  got something contradictory information about H100 cards running massive PIC simulations. 
Thank you you for your help.

Paolo

**Comment by ax3l** [comment, marked as answer]
Hi @TomPaolo,

Thank you for your question!
Yes, we run on A100 and V100 regularly and the performance difference is about 2x.
- V100: https://www.nvidia.com/en-us/data-center/v100/
- A100: https://www.nvidia.com/en-us/data-center/a100/
- H100: https://www.nvidia.com/en-us/data-center/h100/

So far, some of our colleagues were able to run tests with other AMReX codes that also saw again decent speedup from using H100 over A100.

We are currently in the discussion with partners to get our hands for WarpX tests on H100s. From the specs we see so far, we do expect somewhere between 1.5-3x with them. With the high memory-bandwidth and **FP64** FLOP/s on a H100 that helps simulations significantly, H100 is the specialized current Hopper card for HPC.
Some people comment that Hopper would be fully AI/ML centric. Looking at the [H200](https://www.nvidia.com/en-us/data-center/h200/) product (that has more tensor cores), it still has the same FP64 FLOP/s as [H100](https://www.nvidia.com/en-us/data-center/h100/) but 1.8x the memory capacity and an 1.4x increase in memory bandwidth - each things that also help with a PIC simulation's memory-limited algorithms. 

Here is what I look out for when procuring GPUs:
- There are different `Form Factor`s for A100 and V100 - each are tailored to specific applications and interconnects
- Check the advertised `FP64` (not the `FP64 Tensor Core`) *and* `GPU memory` *and* `GPU memory bandwidth`
  - For PIC simulations, I prioritize in this order:
    1. memory bandwidth (faster is better: PIC is memory-bound for the core algorithms)
    2. memory size (more is better: more simulation space fits per card procured)
    3. FP64: (more is better: faster time-to-solution)

If you look in the above specs, avoid the `H100 PCIe` as it has the same `GPU memory bandwidth` as a A100. But `H100 NVL` has nearly 2x faster memory and `H100 SXM` 1.67x faster memory than a A100 - that should give a good speedup.

If you have a system to testing, let us know what you find. We can definitely also keep you posted once we have done some tests on H100s.
